### PR TITLE
Add mail

### PR DESCRIPTION
**Pull request by holywyvern**
# Summary

- Added sidekiq as a gem
- Created a notification job
- Configure jobs to use sidekiq as it's queue
- Added the sending mechanism when an user rents a book
- Configured mail server options

# Trello Card

https://trello.com/c/3YgnxdQ8

### CODE REVIEW

File: .gitgnore [hunk 1]
```diff
@@ -1 +1,2 @@
 test
+.env
```
Comment (juanisierra) [inline .gitgnore]: This file is not correctly named so it won't work

File: app/mailers/application_mailer.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
 class ApplicationMailer < ActionMailer::Base
-  default from: 'from@example.com'
+  default from: ENV['no-reply@training.wolox.com.ar']
```
Comment (juanisierra) [inline app/mailers/application_mailer.rb]: Why is the env variable named like a mail?

File: config/application.rb [hunk 3]
```diff
@@ -15,5 +15,14 @@ class Application < Rails::Application
     # Application configuration can go into files in config/initializers
     # -- all .rb files in that directory are automatically loaded after loading
     # the framework and any gems in your application.
+    config.active_job.queue_adapter = :sidekiq
+    config.action_mailer.smtp_settings = {
+      domain: ENV['MAILER_DOMAIN'],
```
Comment (juanisierra) [inline config/application.rb]: Try using rails secrets for this variables

**Review by juanisierra — CHANGES_REQUESTED** [review]

File: .gitgnore (hunk 1 above)
Comment (juanisierra) [reply .gitgnore]: Remove this file please

File: config/secrets.yml [hunk 4]
```diff
@@ -1,2 +1,24 @@
 development:
-    mailer_domain: <%= ENV["MAILER_DOMAIN"] %>
\ No newline at end of file
+    mailer_domain: <%= ENV["MAILER_DOMAIN"] %>
```
Comment (juanisierra) [inline config/secrets.yml]: You can create a default environment with all variables in common and then inherit it from the others

File: Gemfile [hunk 5]
```diff
@@ -46,6 +46,11 @@ gem 'wor-paginate'
 # Background threads
 gem 'sidekiq'
 
+# Template engine
+gem 'slim'
+
+gem 'dotenv-rails', groups: [:development, :test]
```
Comment (juanisierra) [inline Gemfile]: Put it inside of the group defined below here so it's better organized

**Review by juanisierra — CHANGES_REQUESTED** [review]

File: app/controllers/api/v1/users/rents_controller.rb [hunk 6]
```diff
@@ -1,17 +1,14 @@
...
-        include Wor::Paginate
-
+      class RentsController < ApiController
         def index
           render_paginated Rent.where(user_id: params[:user_id])
         end
 
         def create
           rent = Rent.create(rent_params)
+          RentNotificationsJob.perform_later(rent)
```
Comment (leamotta) [inline app/controllers/api/v1/users/rents_controller.rb]: It is better to just send the rent id here, that way you don't serialize the whole entity into redis, using potentially a lot of memory.

File: config/secrets.yml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+    mailer_address: <%= ENV["MAILER_ADDRESS"] %>
+    mailer_port: <%= ENV["MAILER_PORT"] %>
+    mailer_auth: <%= ENV["MAILER_AUTH"] %>
+    mailer_email: <%= ENV["MAILER_EMAIL"] %>
+development:
+    <<: *default
+production:
+    <<: *default
+test:
+    <<: *default
```
Comment (leamotta) [inline config/secrets.yml]: 🤔  this file should already be created at the start of the project. You need to at least have the `secret_key_base` here
You can set it up like this

development:
&nbsp;  secret_key_base: xxxxxxx

test:
&nbsp;  secret_key_base: xxxxxxx

production:
&nbsp;  secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>

You can read about it here for example https://medium.com/@michaeljcoyne/understanding-the-secret-key-base-in-ruby-on-rails-ce2f6f9968a1

**Review by leamotta — DISMISSED** [review]

**Review by juanisierra — APPROVED** [review]

**Pull request merged**